### PR TITLE
Support build against sle12sp5

### PR DESCRIPTION
**Pull request by nick-wang**
Support build with make_build undefined and enable GNU c99.

Backport clean func sd_bus_message_unrefp and sd_bus_flush_close_unrefp

### CODE REVIEW

File: rpm/sapwmp.spec [hunk 1]
```diff
@@ -61,7 +61,13 @@ Configuration and utilities for collecting SAP processes under control group to
 %build
 ./autogen.sh
 %configure
+# make_build not defined in SLE12, so need conditions to check
```
Comment (Werkov) [inline rpm/sapwmp.spec:64]: The [wiki](https://en.opensuse.org/openSUSE:Specfile_guidelines#Parallel_make) says that %make_build should be available in >= SLE12 SP3. We really don't want to go back that much into the past.
If the wiki's wrong and even SLE12 SP5 needs this, I'm fine with the change.

File: src/main.c [hunk 2]
```diff
@@ -19,6 +19,18 @@
 
 #define _cleanup_(x) __attribute__((cleanup(x)))
 
+#ifndef _SD_DEFINE_POINTER_CLEANUP_FUNC
```
Comment (Werkov) [inline src/main.c:24]: The macro comes from libsystemd's `_sd-common.h` private header.
I'd add a comment with that path: a) gives credit, b) provides hints where to look at if systemd changes it in the future.

**Review by Werkov — CHANGES_REQUESTED** [review]
Thanks. Just nitpicks.

File: src/main.c (hunk 2 above)
Comment (nick-wang) [reply src/main.c:24]: Got it. Added in the new commit.

File: rpm/sapwmp.spec (hunk 1 above)
Comment (nick-wang) [reply rpm/sapwmp.spec:64]: In reality, this macro is not existed in SLE12SP5... rpm-4.11.2-16.21.1.x86_64

```
# rpmbuild --showrc |grep make
      echo "Running automake --gnu ${am_opt} ..."; 
      %{__automake} --add-missing --gnu ${am_opt}; 
-14: __automake	automake
-14: __make	/usr/bin/make
	    yes "" | make oldconfig 
	    make dep 
-14: make_install	make install DESTDIR=%{?buildroot}
-14: makeinstall	make DESTDIR=%{?buildroot:%{buildroot}} install
-14: perl_make_install	make DESTDIR=$RPM_BUILD_ROOT install_vendor
```

**Comment by nick-wang** [comment]
Still have two rpmlint issues need to be fixed to complete the SLE12SP5 build:
```
[   20s] sapwmp.x86_64: E: permissions-file-setuid-bit (Badness: 10000) /usr/lib/sapwmp/sapwmp-capture is packaged with setuid/setgid bits (04750)
[   20s] If the package is intended for inclusion in any SUSE product please open a bug
[   20s] report to request review of the program by the security team
[   20s] 
[   20s] sapwmp.x86_64: E: non-position-independent-executable (Badness: 10000) /usr/lib/sapwmp/sapwmp-capture
[   20s] This executable must be position independent.  Check that it is built with
[   20s] -fPIE/-fpie in compiler flags and -pie in linker flags.

```
`permissions-file-setuid-bit` need security team to involve.

File: rpm/sapwmp.spec (hunk 1 above)
Comment (Werkov) [reply rpm/sapwmp.spec:64]: Ack.

File: configure.ac [hunk 3]
```diff
@@ -1,8 +1,12 @@
 AC_INIT([sdcapture], 1.0)
 AM_INIT_AUTOMAKE([subdir-objects])
 AC_PROG_CC
+AC_PROG_CC_C99
 AC_CONFIG_FILES(Makefile)
 
+AC_SUBST([CFLAGS], [-fPIE])
+AC_SUBST([LDFLAGS], [-pie])
```
Comment (Werkov) [inline configure.ac:8]: Ack.

**Comment by Werkov** [comment]
Thanks, LGTM (the rpmlint has to be resolved out of the package itself) -> merging.

**Comment by nick-wang** [comment]
@Werkov Thanks for the review and information!

**Pull request merged**